### PR TITLE
Bring release management doc up to date

### LINKED ISSUE #23: Revise release management guide
### Descriptive summary

A lot has changed, and should change going forward. For example we should **not** be moving a `staging` tag around to trigger CI builds to staging. That's not a good practice. We also don't **require** that one use the git-flow plugin for creating and managing release branches. So perhaps rewrite this in favor of using plain git commands. This would make it a bit easier for teaching the process to others.

### PR DESCRIPTION
**Pull request by mcritchlow**
Closes #23 

Changes proposed in this pull request:
- add regular `git` command alternatives to `git-flow`
- deprecate using the staging tag for new apps
- remove references to JIRA and replace w/ GitHub
- remove references to Bamboo and replace w/ Jenkins
- add instruction to create GitHub release

@ucsdlib/developers - please review. I'd especially appreciate thoughts on the addition of the standard git commands (such as merge vs rebase) and the inclusion of instructions to create a GitHub release. Thanks!

### CODE REVIEW

File: protocol/release_management.md [hunk 1]
```diff
@@ -22,46 +20,70 @@ git checkout develop
...
 git push --tags
 ```
-2. Release Manager assigns to Operations for Production release.
-3. When application is released to Production Operations assigns Deploy ticket
+**or**
+```
+# update develop branch if changes were made in release branch
+git checkout develop
+git rebase release/1.2.3
+
```
Comment (hweng) [inline protocol/release_management.md]: @mcritchlow  Should it be pushed to github `git push origin develop`?

**Comment by hweng** [comment]
@mcritchlow Overall looks :+1: to me except for the above question.

File: protocol/release_management.md (hunk 1 above)
Comment (mcritchlow) [reply protocol/release_management.md]: ah yes! good point. I'll update that now, thanks :)

Comment (mcritchlow) [reply protocol/release_management.md]: updated in ab3b7ce thanks for catching that @hweng 👍

**Comment by hweng** [comment]
:+1:

**Comment by lsitu** [comment]
👍

**Pull request merged**